exonum-java-binding/doc/Migration_guide_0.9.md

```diff
@@ -0,0 +1,183 @@
+# Migrating Java Services to 0.9
+
+The flagship feature of Exonum Java 0.9 release is support for dynamic services&nbsp;—
+services that can be added to the blockchain network after it has been started;
+and multiple instances of the same service.
+
+This guide provides instructions on updating existing services to be compatible with the new
+and updated features.
+
+## Dynamic Services Overview
+
+Dynamic Java services are packaged in an _artifact_: a JAR archive with some metadata. An Exonum
+artifact has a runtime-specific _name_. Java services must use name in the format 
+'groupId:artifactId:version', where each component of the name corresponds
+to a Maven project coordinate, for example: 'com.exonum.example:timestamping:1.0.2' is a valid
+artifact name. Rust service names are usually in format '<crate-name>:<version>'.
+The format of the metadata has not changed since 0.6.0.
+
+Artifacts are _deployed_ to the Exonum blockchain network; and then used to instantiate
+_service instances_. A single artifact can be used to create many instances. Each instance
+has a _name_ (ex `Service#getName`) and a numeric _identifier_ (ex `Service#getId`).
+An instance name is specified by the network administrators during service instantiation;
+an id is assigned by the framework automatically and can be [queried][service-id-lc-operation]
+using the light client.
+
+[service-id-lc-operation]: todo
+
+## Update the Service
+
+The [`Service`][service] implementation must be updated to support multiple instances;
+new initialization and reconfiguration procedures.
+
+1. Remove the hardcoded service id and name. Add a constructor parameter 
+`ServiceInstanceSpec instanceSpec` which provides instance name
+and id.
+    - If your service extends
+    [`com.exonum.binding.core.service.AbstractService`][abstract-service] —
+    pass the `instanceSpec` to the superclass constructor and use provided `AbstractService#getId`,
+    `getName`, and `getInstanceSpec` methods
+    - If it implements `Service` directly — save the needed information to fields.
```

Comment:
```suggestion
    - If it implements `Service` directly — save the needed information to the fields.
```

Reply:
Just to clarify: _some_ fields (we don't say to which fields or how many of them or of which type — that's up to the implementor), but _the_ fields defined in this class (what is referred to by 'it' preposition). Is it 'the'?